src/front/glsl/parser.rs

```diff
@@ -244,10 +244,25 @@ pomelo! {
     function_call ::= function_call_or_method(fc) {
         match fc.kind {
             FunctionCallKind::TypeConstructor(ty) => {
-                let h = extra.context.expressions.append(Expression::Compose {
-                    ty,
-                    components: fc.args.iter().map(|a| a.expression).collect(),
-                });
+                let h = if fc.args.len() == 1 {
+                    let kind = match extra.module.types[ty].inner {
```

Comment:
could we use the existing `scalar_kind()` method here? See https://github.com/gfx-rs/naga/blob/9041cd4b10ec6105d9041d103fe0a78022c3472c/src/proc/mod.rs#L54

Reply:
Ahh, of course - hadn't spotted that one. Guess need to do a new PR :)